### DISCUSSION TITLE
Not sure if my browser implementation is correct ...

### DISCUSSION THREAD

**Discussion by fazulk** [panva/jose#181] [Q&A] [answered]
Trying to use jose in the chrome, and not having luck. I don't think this a bug, but more of user error on some level. 

```
import { SignJWT } from 'jose/jwt/sign'

const privateKey = `
-----BEGIN RSA PRIVATE KEY-----
MIIBOAIBAAJAYpRKQKljpX-BLAH
-----END RSA PRIVATE KEY-----
`

const jwt = await new SignJWT({ 'urn:example:claim': true })
      .setProtectedHeader({ alg: `RS256` })
      .sign(privateKey)

console.log(jwt)
```

in chrome error is :
```
webcrypto.js:4 Uncaught (in promise) TypeError: Right-hand side of 'instanceof' is not an object
    at isCryptoKey (webcrypto.js:4)
    at getCryptoKey (get_sign_verify_key.js:3)
    at sign (sign.js:6)
    at FlattenedSign.sign (sign.js:63)
    at CompactSign.sign (sign.js:11)
    at SignJWT.sign (sign.js:17)
    at createToken (tokens.js?t=1618511376485:48)
    at sample_user.js?t=1618511376485:16
  ```

I also tried using the browser only version of the package, but resulted in a export error. I'm using vite as my developement env, which uses esbuild. Any examples of browser use would be greatly appreciated, if they are different from the docs.

**Comment by panva** [comment, marked as answer]
@fazulk the sign argument is meant to be a [KeyLike](https://github.com/panva/jose/blob/v3.11.5/docs/types/types.keylike.md) (type alias), in the browser this means all asymmetric keys need to be CryptoKey instances, e.g. imported via https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/importKey

**Reply by panva** [reply]
Nevertheless i'd be interested in what the browser in question is, since this seems to mean the globalThis CryptoKey is not defined.

**Reply by panva** [reply]
Which may also be caused by not being on a [SecureContext](https://developer.mozilla.org/en-US/docs/Web/API/CryptoKey).

**Reply by fazulk** [reply]
@panva exactly what it was. Running in Chrome 90. As soon as i turned on SSL everything worked like a charm. Thanks!
